src/events/crawler/lib/countryLevels.js

```diff
@@ -0,0 +1,76 @@
+import * as fs from './fs.js';
+import * as geography from './geography.js';
+
+export function isId(str) {
+  if (!str) return false;
+  const [level, code] = str.split(':');
+  return ['id0', 'id1', 'id2', 'id3'].includes(level) && !!code;
+}
+
+export function getIdFromLocation(location) {
+  const smallestLocationStr = geography.getSmallestLocationStr(location);
+  return isId(smallestLocationStr) ? smallestLocationStr : null;
+}
+
+export function splitId(id, lower = false) {
+  if (!isId(id)) {
+    console.error(`  ❌ Wrong id: ${id}`);
+    return {};
+  }
+
+  const id_ = lower ? id.toLowerCase() : id;
+  const [level, code] = id_.split(':');
+  return { level, code };
+}
+
+function splitId3(code, lower = false) {
+  const code_ = lower ? code.toLowerCase() : code;
+  const [country, state] = code_.split('-');
+  if (!country || !state) {
+    console.error(`  ❌ Wrong id3: ${code}`);
+  }
+  return { country, state };
+}
+
+function cleanProperties(feature) {
+  const prop = feature.properties;
+  feature.properties = {
+    name: prop.name,
+    population: prop.population,
+    wikidata_id: prop.wikidata_id,
+    ne_id: prop.ne_id,
+    cl_id: prop.cl_id
```

Comment:
Is there a good reason to have these as abbreviations? I'm guessing NE is natural earth, not sure what CL is. A casual reader would likely be lost here.